File: array/circular_counter/circular_counter.js
let a = ['1', '2', '3', '4', '5', '6', '7', '8', '9']

function circCounter(arr, skip){
    let index = 0
    while (arr.length > 0){
        index = (index+skip-1) % arr.length
        console.log(arr.splice(index, 1)[0])
    }
}

circCounter(a, 3)

File: array/circular_counter/circular_counter.rb
a = %w[1 2 3 4 5 6 7 8 9]

def circ_counter(arr, skip)
  index = 0
  until arr.empty?
    index = (skip + index - 1) % arr.length
    puts arr.delete_at(index)
  end
end

circ_counter(a, 3)
